src/js/hooks/useUserHasPermissions.js

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useMemo } from 'react';
+
+import _ from 'lodash';
+import { useSelector } from 'react-redux';
+
+import RoleType from 'consts/roleType';
+
+/**
+ * Returns true or false if current logged-in user has all the roles specified in the parameters.
+ * The function checks if user has either global or location based roles.
+ * @param minRequiredRole {RoleType} - minimum required role
+ * @param supplementalRoles {RoleType[]} [d=[]] - list of supplemental required roles
+ * @returns {boolean}
+ */
+const useUserHasPermissions = ({ minRequiredRole, supplementalRoles = [] }) => {
```

Comment:
Should this be a replacement for the `permisionUtil` that we already have? Is there a case that we want to have both? I think we should have only one thing if that's possible.